<a href="https://colab.research.google.com/github/Renad110/IT326/blob/main/Phase2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import pandas as pd
url = "https://raw.githubusercontent.com/Renad110/IT326/refs/heads/main/Dataset/Raw_dataset/bank-additional-full.csv"
df = pd.read_csv(url, sep=';')

🔹 Statistical & Data Overview Analysis:
In this section, we explore the dataset by showing its structure, statistical summary, missing values, and outliers. This helps us understand data quality and what preprocessing might be needed later.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41188 entries, 0 to 41187
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   age             41188 non-null  int64  
 1   job             41188 non-null  object 
 2   marital         41188 non-null  object 
 3   education       41188 non-null  object 
 4   default         41188 non-null  object 
 5   housing         41188 non-null  object 
 6   loan            41188 non-null  object 
 7   contact         41188 non-null  object 
 8   month           41188 non-null  object 
 9   day_of_week     41188 non-null  object 
 10  duration        41188 non-null  int64  
 11  campaign        41188 non-null  int64  
 12  pdays           41188 non-null  int64  
 13  previous        41188 non-null  int64  
 14  poutcome        41188 non-null  object 
 15  emp.var.rate    41188 non-null  float64
 16  cons.price.idx  41188 non-null  float64
 17  cons.conf.idx   41188 non-null 

The dataset contains 41188 rows and 21 columns.  
Most of the attributes are either categorical or numeric,  
which means both types of analysis will be needed.

In [6]:
df.describe()

,age,duration,campaign,pdays,previous,emp.var.rate,cons.price.idx,cons.conf.idx,euribor3m,nr.employed
count,41188.00000,41188.000000,41188.000000,41188.000000,41188.000000,41188.000000,41188.000000,41188.000000,41188.000000,41188.000000
mean,40.02406,258.285010,2.567593,962.475454,0.172963,0.081886,93.575664,-40.502600,3.621291,5167.035911
std,10.42125,259.279249,2.770014,186.910907,0.494901,1.570960,0.578840,4.628198,1.734447,72.251528
min,17.00000,0.000000,1.000000,0.000000,0.000000,-3.400000,92.201000,-50.800000,0.634000,4963.600000
25%,32.00000,102.000000,1.000000,999.000000,0.000000,-1.800000,93.075000,-42.700000,1.344000,5099.100000
50%,38.00000,180.000000,2.000000,999.000000,0.000000,1.100000,93.749000,-41.800000,4.857000,5191.000000
75%,47.00000,319.000000,3.000000,999.000000,0.000000,1.400000,93.994000,-36.400000,4.961000,5228.100000
max,98.00000,4918.000000,56.000000,999.000000,7.000000,1.400000,94.767000,-26.900000,5.045000,5228.100000


The table above provides a five-number summary and basic statistics for the numerical columns.
For example, we can see that the average of most numerical attributes, such as `age` and `duration`,
shows moderate variation between the minimum and maximum values, which helps identify data spread and potential outliers.


In [7]:
missing_values = df.isna().sum()
print("Missing values in each column:")
print(missing_values)

Missing values in each column:
age               0
job               0
marital           0
education         0
default           0
housing           0
loan              0
contact           0
month             0
day_of_week       0
duration          0
campaign          0
pdays             0
previous          0
poutcome          0
emp.var.rate      0
cons.price.idx    0
cons.conf.idx     0
euribor3m         0
nr.employed       0
y                 0
dtype: int64


Missing Values Analysis:
We checked for missing values in each column. The output shows that all columns have zero missing values,
which means the dataset is complete and no imputation is required at this stage.


In [9]:
from scipy.stats import zscore
# sample dataset
data = df['age']

z_scores = zscore(data)
threshold = 2

outliers = data[abs(z_scores) > threshold]
print("Outliers:", outliers)


Outliers: 8645     61
15798    19
24110    18
27713    70
27715    61
         ..
41174    62
41178    62
41179    64
41183    73
41187    74
Name: age, Length: 985, dtype: int64


Outliers Detection: We used the Z-Score method on the age column to detect unusual values.
This method can also be applied to other numeric columns to find outliers.
The detected extreme values will be handled later during preprocessing.

In [10]:
df['y'].value_counts()

,count
y,
no,36548
yes,4640


Class Label Count: We used value_counts() to check the distribution of the class label.
The dataset contains 36,548 samples labeled “no” and 4,640 samples labeled “yes”,
indicating that the dataset is imbalanced.
This imbalance will be considered later during preprocessing to improve model performance.